src/main/java/com/commercetools/sync/producttypes/ProductTypeSync.java

```diff
@@ -223,7 +225,15 @@ private void handleError(@Nonnull final String errorMessage, @Nullable final Thr
                 removeAndKeepTrackOfMissingNestedAttributes(newProductType, keyToIdCache))
             .map(draftWithoutMissingRefAttrs -> referenceResolver
                 .resolveReferences(draftWithoutMissingRefAttrs)
-                .thenCompose(resolvedDraft -> syncDraft(oldProductTypeMap, resolvedDraft)))
+                .thenCompose(resolvedDraft -> syncDraft(oldProductTypeMap, resolvedDraft))
+                .exceptionally(completionException -> {
```

Comment:
I am not sure from that type `CompletionException` from a library design perspective, it's itself related to the java futures because the task is running concurrently and exceptions happening in the future stages, so the exception is wrapped by CompletionException.class always, what about returning getCause, I would suggest some investigation on that: 

From the https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletionStage.html:
``` 
 In all other cases, if a stage's computation terminates abruptly with an 
(unchecked) exception or error, then all dependent stages requiring its completion complete exceptionally as well, with a 
CompletionException holding the exception as its cause. If a stage is dependent on both of two stages, and both 
complete exceptionally, then the CompletionException may correspond to either one of these exceptions. If a stage is 
dependent on either of two others, and only one of them completes exceptionally, no guarantees are made about whether 
the dependent stage completes normally or exceptionally. 
```

Reply:
That's what I have explained two weeks ago, turn out our conclusion is to get rid of returning .getCause(). Anyway when we put the raw exception into handleError(..) method, it doesn't matter what the actual exception class it wraps.

Reply:
As suggested, it now returns CompletionException.getCause().